students/jesse_miller/session04/mailroom.py

```diff
@@ -0,0 +1,194 @@
+    print("-"*80)
+    print("{:17} | {:<20} | {:<15} | {:<19}".format(headers[0], headers[1], \
+    headers[2], headers[3]))
+    print("-"*80)
+
+    for k, v in donors.items():
+        total = (sum(v))
+        times = (len(v))
+        avg = (sum(v) / len(v))
+        summary.append([k, total, times, avg])
+    summary.sort(key=lambda d: d[1], reverse=True)
+    for x in summary:
+        print("{:17} | ${:<18,.2f} | {:<15} |  ${:<17,.2f}".format(x[0], x[1], \
+        x[2], x[3]))
+    print("-"*80)
+    print("")
+
+
+def goodbye():
+    '''
+    Gracefully exits
+    '''
+    print("Goodbye!")
+    sys.exit()
+
+
+def donor_list():
+    '''
+    This when done properly, will print the list of donor names
+    '''
+    print("\n")
+    print("-" * 15)
+    print("List of donors: ")
+    print("-" * 15)
+    for donor in donors:
+        print(donor)
+    print("-" * 15)
+    print("\n")
+
+
+def donor_mail():
+    """
+    This section allows the user to mail a donor
+    """
+    current_donor = ""
+    donor_list()
+    current_donor = str(input("Who would you like to mail: "))
+    if current_donor in donors:
+        print("")
+        mail_send(current_donor)
+    else:
+        donor_add(current_donor)
+
+
+def donor_add(current_donor):
+    """
+    This allows addition of new donors
+    """
+    if current_donor not in donors:
+        donors[current_donor] = []
+    d_num = int(input("How many donations were made: "))
+    while d_num > 0:
+        new_don = float(input("Enter their donation: "))
+        donors[current_donor].append(new_don)
+        d_num -= 1
+    mail_send(current_donor)
+
+
+def donor_del():
+    """
+    This section allows the user to delete a donor
+    """
+    del_donor = str(input("Enter the name of the donor to remove: "))
+    del donors[del_donor]
+    donor_list()
+
+
+def mail_send(current_donor):
+    """
+    This is the meat of the send process
+    """
+    donor_math = donors[current_donor]
+    donor_total = sum(donor_math)
+    donor_avg = len(donor_math)
+    mail = ("Hello {}, \n"
+            "\n"
+            "We are writing to thank you for you generous donation\n"
+            "to our foundation.  Your contributions for the year \n"
+            "total ${:,.2f} in {} disbursements. \n"
+            "\n"
+            "Again, the foundation thanks you for your support, \n"
+            "and we hope to remain in contact with you in this new \n"
+            "year.\n"
+            "\n"
+            "Sincerely, \n"
+            "Ecumenical Slobs LLC \n".format(current_donor, donor_total, \
+            donor_avg))
+    print(mail)
+
+
+def mail_file():
+    """
+    Hopefully, this makes directories and files on first run for the listed
+    donors.  Hopefully
+    """
+    path = os.getcwd()
+    for k in donors:
+        current_donor = ""
+        current_donor = k
+        donor_math = donors[current_donor]
+        donor_total = sum(donor_math)
+        donor_avg = len(donor_math)
```

Comment:
You have this same summation of the donor stuff further up the code.  Might that be a good candidate for a function to avoid duplication?